packages/shared/test/porter.test.ts

```diff
@@ -117,3 +159,103 @@ describe('PorterClient', () => {
     );
   });
 });
+
+describe('PorterClient Signing', () => {
+  beforeAll(async () => {
+    await initialize();
+  });
+
+
+  describe('signUserOp', () => {
+    const mockPackedUserOp = {
+      sender: '0x1234',
+      nonce: '0x1',
+      factory: '0x0',
+      factoryData: '0x',
+      callData: '0xabc',
+      callGasLimit: '0x20000',
+      verificationGasLimit: '0x15000',
+      preVerificationGas: '0x1000',
+      maxFeePerGas: '0xabc',
+      maxPriorityFeePerGas: '0x123',
+      paymaster: '0x0',
+      paymasterVerificationGasLimit: '0x0',
+      paymasterPostOpGasLimit: '0x0',
+      paymasterData: '0x',
+      signature: '0x',
+    };
+
+    it('should successfully sign a UserOperation', async () => {
+      mockSignUserOp(true);
+      const porterClient = new PorterClient(fakePorterUris[2]);
+      const result = await porterClient.signUserOp(
+        {
+          '0x1234': JSON.stringify(mockPackedUserOp),
+          '0xabcd': JSON.stringify(mockPackedUserOp)
+        },
+        2,
+        {
+          optimistic: true,
+          returnAggregated: true
+        }
+      );
+
+      expect(result).toEqual({
+        digest: '0x1234',
+        aggregatedSignature: '0x90ab0xijkl', // Combined signatures in sorted order
+        signingResults: {
+          '0x1234': ['0x5678', '0x90ab'],
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {},
+      });
+    });
+
+    it('should handle UserOperation signing failures', async () => {
+      mockSignUserOp(false);
+      const porterClient = new PorterClient(fakePorterUris[2]);
+
+      await expect(
+        porterClient.signUserOp(
+          {
+            '0x1234': JSON.stringify(mockPackedUserOp),
+            '0xabcd': JSON.stringify(mockPackedUserOp)
+          },
+          2,
+          {
+            optimistic: true,
+            returnAggregated: true
+          }
+        )
+      ).rejects.toThrow('Porter returned bad response: 400 - ');
+    });
+
+    it('should handle signature decoding errors in UserOperation signing', async () => {
+      // Mock a response with invalid base64 signature
+      mockSignUserOp(true, true);
+      const porterClient = new PorterClient(fakePorterUris[2]);
+      const result = await porterClient.signUserOp(
+        {
+          '0x1234': JSON.stringify(mockPackedUserOp),
+          '0xabcd': JSON.stringify(mockPackedUserOp)
+        },
+        2,
+        {
+          optimistic: true,
+          returnAggregated: true
+        }
+      );
+
+      expect(result).toEqual({
+        digest: '0x1234',
+        aggregatedSignature: '0xijkl', // Only valid signature included
+        signingResults: {
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {
+          '0x1234': 'Failed to decode signature: Invalid character',
```

Comment:
Need to better understand this - an aggregated signature is produced when there wasn't a threshold of signatures?